### PR TITLE
Add `catboost` to the third-party integration tests

### LINKED ISSUE #15397: [FEA] Improve support or failure modes for numpy and other libraries with C APIs in cudf.pandas
Currently we proxy numpy.ndarray to ensure that cupy arrays are produced instead where possible. However, this behavior only partially addresses the possible ways to interact with numpy. Similarly to how we handle pandas in cudf.pandas, we may want to install a proxied library for numpy itself in such cases to ensure that we get the desired behavior.

Unfortunately, this is far more challenging with numpy than with pandas due to the fact that numpy exposes a C API. This issue is not unique to numpy, but is also present for other libraries (like torch) that rely on being able to translate Python objects from standard vocabulary types like numpy arrays down to C representations to leverage in their own C code. In general, our approach for proxying is incompatible with code that relies on converting Python objects into their C representations since we cannot mimic the latter, only the former.

We should collect cases where we observe failures like this and see if we can, at minimum, improve the ways in which the code fails. If possible, we can also try to come up with more robust strategies for consuming libraries to use such that they won't accidentally go down such bad code paths with cudf.pandas objects.

In an ideal world, we would come up with a solution that actually enables support for such use cases, but at present I don't see how we could manage doing so without doing something crazy like hooking every function call with `sys.setprofile`, and even if that worked (I'm not sure that it would) the cure might be worse than the disease because we'd most likely slow down every function call in Python enough to overcome any gains from using cudf.pandas.

### PR DESCRIPTION
**Pull request by Matt711**
## Description

Closes #15397 
## Checklist
- [ ] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

File: .github/workflows/pr.yaml [hunk 1]
```diff
@@ -37,6 +37,7 @@ jobs:
       - unit-tests-cudf-pandas
       - pandas-tests
       - pandas-tests-diff
+      - third-party-integration-tests-cudf-pandas
```
Comment (Matt711) [inline .github/workflows/pr.yaml:40]: Ignore the CI changes. I added them to show the third-party integration tests were passing. I'll delete them once this PR is approved.

Comment (Matt711) [reply .github/workflows/pr.yaml:40]: Sorry for the ping

File: python/cudf/cudf_pandas_tests/third_party_integration_tests/dependencies.yaml [hunk 2]
```diff
@@ -244,6 +251,16 @@ dependencies:
...
+  test_catboost:
+    common:
+      - output_types: conda
+        packages:
+          - numpy
+          - scipy
+          - scikit-learn
+          - pip
+          - pip:
+            - catboost
```
Comment (mroeschke) [inline python/cudf/cudf_pandas_tests/third_party_integration_tests/dependencies.yaml:263]: Do we need to install this via pip? Appears catboost is on conda forge https://catboost.ai/en/docs/installation/python-installation-method-conda-install

Comment (bdice) [reply python/cudf/cudf_pandas_tests/third_party_integration_tests/dependencies.yaml:263]: Please use conda if possible.

**Pull request merged**